pkg/controller/windowsmachineconfig/windowsmachineconfig_controller.go

```diff
@@ -100,54 +122,154 @@ func (r *ReconcileWindowsMachineConfig) Reconcile(request reconcile.Request) (re
+}
+
+// chooseRandomNode chooses one of the windows nodes randomly. The randomization is coming from golang maps since you
+// cannot assume the maps to be ordered.
+func chooseRandomNode(windowsVMs map[types.WindowsVM]bool) types.WindowsVM {
+	for windowsVM := range windowsVMs {
+		return windowsVM
+	}
+	return nil
+}
+
+// deleteWindowsVMs deletes the required number of Windows VMs from the cluster and returns a bool indicating the
+// status of deletion. This method will return false if any of the VMs fail to get deleted.
+// TODO: This method should return a slice of errors that we collected.
+//		Jira story: https://issues.redhat.com/browse/WINC-266
+func (r *ReconcileWindowsMachineConfig) deleteWindowsVMs(count int) bool {
+	var errs []error
+	// From the list of Windows VMs choose randomly count number of VMs.
+	for i := 0; i < count; i++ {
+		// Choose of the Windows worker nodes randomly
+		vmTobeDeleted := chooseRandomNode(r.windowsVMs)
+		if vmTobeDeleted.GetCredentials() == nil {
+			errs = append(errs, errors.New("One of the VM deletions failed, will reconcile..."))
+			continue
+		}
+
+		// Get the instance associated with the Windows worker node
+		instancedID := vmTobeDeleted.GetCredentials().GetInstanceId()
+		if len(instancedID) == 0 {
+			errs = append(errs, errors.New("One of the VM deletions failed, will reconcile..."))
+			continue
+		}
+
+		// Delete the Windows VM from cloud provider
+		log.Info(fmt.Sprintf("deleting the Windows VM: %s", instancedID))
+		if err := r.cloudProvider.DestroyWindowsVM(instancedID); err != nil {
+			log.Error(err, "error while deleting windows VM %s", instancedID)
+			errs = append(errs, errors.Wrap(err, "One of the VM deletions failed, will reconcile..."))
+		}
+		delete(r.windowsVMs, vmTobeDeleted)
+	}
+
+	// If any of the Windows VM fails to get deleted consider this as a failure and return false
+	if len(errs) > 0 {
+		return false
+	}
+	return true
+}
+
+// createWindowsVMs creates the required number of windows Windows VM and configure them to make
+// TODO: This method should return a slice of errors that we collected.
+//		Jira story: https://issues.redhat.com/browse/WINC-266
+func (r *ReconcileWindowsMachineConfig) createWindowsWorkerNodes(count int) bool {
+	var errs []error
+	for i := 0; i < count; i++ {
+		// Create Windows VM in the cloud provider
+		createdVM, err := r.cloudProvider.CreateWindowsVM()
+		if err != nil {
+			errs = append(errs, errors.Wrap(err, "error creating windows VM"))
+			log.Error(err, "error while creating windows VM")
+		}
+		log.V(5).Info(fmt.Sprintf("created the Windows VM: %s",
+			createdVM.GetCredentials().GetInstanceId()))
+
+		// Make the Windows VM a Windows worker node.
+		nc := nodeconfig.NewNodeConfig(r.k8sclientset, createdVM)
+		if err := nc.Configure(); err != nil {
+			// TODO: Unwrap to extract correct error
+			errs = append(errs, errors.Wrap(err, "configuring Windows VM failed"))
+			log.Error(err, "configuring Windows VM failed", err)
```

Comment:
Don't log here, just print errs at the end

Reply:
The error will have information related to node.